tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/SearchConfigSpec.scala

```diff
@@ -0,0 +1,596 @@
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.tests.BaseSpec
+import ch.epfl.bluebrain.nexus.tests.Identity.resources.Rick
+import ch.epfl.bluebrain.nexus.tests.iam.types.Permission.{Organizations, Resources}
+import io.circe.Json
+import monix.bio.Task
+import org.scalatest.Assertion
+
+class SearchConfigSpec extends BaseSpec {
+
+  private val orgId    = genId()
+  private val projId1  = genId()
+  private val id1      = s"$orgId/$projId1"
+  private val projects = List(id1)
+
+  private val neuronMorphologyId =
+    "https://bbp.epfl.ch/neurosciencegraph/data/neuronmorphologies/da3b1e42-5f7f-4065-8be3-d2132c219bc2"
+  private val neuronDensityId    =
+    "https://bbp.epfl.ch/neurosciencegraph/data/densities/87221bd5-bd0b-464d-a863-b32d6f62893b"
+  private val traceId            =
+    "https://bbp.epfl.ch/neurosciencegraph/data/traces/8f03a402-f0bb-4114-8a52-e8d3e23949fa"
+  private val layerThicknessId   =
+    "https://bbp.epfl.ch/neurosciencegraph/data/8d3a4d11-cf77-4ac1-ae14-6f2480d0c992"
+  private val boutonDensityId    =
+    "https://bbp.epfl.ch/neurosciencegraph/data/e38cfd23-7028-4fce-82d1-b3beeb1e2fc5"
+
+  // the resources that should appear in the search index
+  private val mainResources  = List(
+    "/kg/search/patched-cell.json",
+    "/kg/search/trace.json",
+    "/kg/search/neuron-morphology.json",
+    "/kg/search/neuron-density.json",
+    "/kg/search/layer-thickness.json",
+    "/kg/search/bouton-density.json"
+  )
+  private val otherResources = List(
+    "/kg/search/article.json",
+    "/kg/search/org.json",
+    "/kg/search/license.json",
+    "/kg/search/activity.json",
+    "/kg/search/protocol.json",
+    "/kg/search/person.json"
+  )
+  private val allResources   = otherResources ++ mainResources
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    val searchSetup = for {
+      _ <- aclDsl.cleanAcls(Rick)
+      _ <- aclDsl.addPermission("/", Rick, Organizations.Create)
+      _ <- adminDsl.createOrganization(orgId, orgId, Rick)
+      _ <- adminDsl.createProject(orgId, projId1, kgDsl.projectJson(path = "/kg/projects/bbp.json", name = id1), Rick)
+
+      _ <- aclDsl.addPermission(s"/$orgId", Rick, Resources.Read)
+      _ <- aclDsl.addPermission(s"/$orgId/$projId1", Rick, Resources.Read)
+
+      _ <- postResource("/kg/search/neuroshapes.json")
+      _ <- postResource("/kg/search/bbp-neuroshapes.json")
+      _ <- allResources.traverseTap(postResource)
+    } yield ()
+
+    searchSetup.accepted
+  }
+
+  "search indexing" should {
+
+    "index all data" in {
+      eventually {
+        deltaClient.post[Json]("/search/query", json"""{"size": 100}""", Rick) { (body, response) =>
+          response.status shouldEqual StatusCodes.OK
+          val sources = Json.fromValues(body.findAllByKey("_source"))
+          sources.asArray.get.size shouldBe mainResources.size
+        }
+      }
+    }
+
+    "index project" in {
```

Comment:
This is not really about indexing there but asserting the property is correctly indexed